core/jvm/src/test/scala/scalaz/zio/ZManagedSpec.scala

```diff
@@ -332,47 +334,73 @@ class ZManagedSpec(implicit ee: org.specs2.concurrent.ExecutionEnv) extends Test
       } yield abs1 must ===(abs2)).use[Any, Nothing, MatchResult[String]](result => ZIO.succeed(result)))
     }
 
-  def timeoutReservation = {
-    val managed = ZManaged.make(clock.sleep(5.seconds))(_ => ZIO.unit)
+  private def timeoutHappy =
     unsafeRun {
-      managed.timeout(1.millisecond).use(res => ZIO.succeed(res must be_===(None)))
+      val managed = ZManaged.make(ZIO.succeed(1))(_ => ZIO.unit)
+      managed.timeout(Duration.Infinity).use(res => ZIO.succeed(res must_=== (Some(1))))
     }
-  }
 
-  def timeoutAcquisition = {
-    val managed = ZManaged.reserve(Reservation(clock.sleep(5.seconds), ZIO.unit))
+  private def timeoutReservation =
     unsafeRun {
-      managed.timeout(1.millisecond).use(res => ZIO.succeed(res must be_===(None)))
+      for {
+        latch   <- Promise.make[Nothing, Unit]
+        managed = ZManaged.make(latch.await)(_ => ZIO.unit)
+        res     <- managed.timeout(Duration.Zero).use(res => ZIO.succeed(res must_=== (None)))
+        _       <- latch.succeed(())
+      } yield res
     }
-  }
 
-  def timeoutAcquisitionAndReservation = {
-    val managed = ZManaged(clock.sleep(1.millisecond) *> ZIO.succeed(Reservation(clock.sleep(5.seconds), ZIO.unit)))
+  private def timeoutAcquisition =
     unsafeRun {
-      managed.timeout(100.millisecond).use(res => ZIO.succeed(res must be_===(None)))
+      for {
+        latch   <- Promise.make[Nothing, Unit]
+        managed = ZManaged.reserve(Reservation(latch.await, ZIO.unit))
+        res     <- managed.timeout(Duration.Zero).use(res => ZIO.succeed(res must_=== (None)))
+        _       <- latch.succeed(())
+      } yield res
     }
-  }
 
-  def timeoutRunFinalizers = {
-    var release = 0
+  private def timeoutAcquisitionAndReservation = {
     val managed = ZManaged(
-      clock.sleep(1.millisecond) *> ZIO.succeed(Reservation(clock.sleep(5.seconds), ZIO.effectTotal(release += 1)))
+      clock.sleep(30.millisecond) *> ZIO.succeed(Reservation(clock.sleep(30.millisecond), ZIO.unit))
     )
     unsafeRun {
-      managed.timeout(100.millisecond).use(res => ZIO.succeed((res must be_===(None)) && (release must be_===(1))))
+      managed.timeout(50.millisecond).use(res => ZIO.succeed(res must be_===(None)))
```

Comment:
These sleeps will be finicky. Even though 50 < 60 theoretically, in reality that won't always be the case. Is there another way we could structure this test?

Reply:
this one is structured the same way in that it expects the execution to take longer than 50 milliseconds.
But after giving this a second thought, this test is always going to be nondeterministic. I'll remove it for now.